docs/release/controllers.md

```diff
@@ -0,0 +1,151 @@
+
+The Flux controllers are 
+[Kubernetes operators](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/),
+each controller has its own Git repository and release cycle.
+
+Controller repositories and their interdependencies:
+
+1. [fluxcd/source-controller](https://github.com/fluxcd/source-controller)
+2. [fluxcd/kustomize-controller](https://github.com/fluxcd/kustomize-controller) (imports `fluxcd/source-controller/api`)
+3. [fluxcd/helm-controller](https://github.com/fluxcd/helm-controller) (imports `fluxcd/source-controller/api`)
+4. [fluxcd/notification-controller](https://github.com/fluxcd/notification-controller)
+5. [fluxcd/image-reflector-controller](https://github.com/fluxcd/image-reflector-controller)
+6. [fluxcd/image-automation-controller](https://github.com/fluxcd/image-automation-controller) (imports `fluxcd/source-controller/api` and `fluxcd/image-reflector-controller/api`)
+
+## API versioning 
+
+The Flux APIs (Kubernetes CRDs) follow the
+[Kubernetes API versioning](https://kubernetes.io/docs/reference/using-api/#api-versioning) scheme.
+
+### Alpha version
+
+An alpha version API e.g. `v1alpha1` is considered experimental and should be used on
+test environments only.
+
+The schema of objects may change in incompatible ways in a later controller release.
+The Custom Resources may require editing and re-creating after a CRD update.
+
+An alpha version API becomes deprecated once a subsequent alpha or beta API version is released.
+A deprecated alpha version is subject to removal after a three months period.
+
+An alpha API is introduced when its proposal reaches the  `implementable` phase in the
+[Flux RFC process](https://github.com/fluxcd/flux2/tree/main/rfcs).
+We encourage users to try out the alpha APIs and provide feedback
+(e.g. on CNCF Slack or in the form of GitHub issues/discussions)
+which is extremely valuable during early stages of development.
+
+### Beta version
+
+A beta version API e.g. `v2beta1` is considered well-tested and safe to use in production.
+
+The schema of objects may change in incompatible ways in a subsequent beta or stable API version.
+The Custom Resources may require editing after a CRD update for which migration instructions will be
+provided as part of the controller changelog.
+
+A beta version API becomes deprecated once a subsequent beta or stable API version is released. 
+A deprecated beta version is subject to removal after a six months period.
+
+### Stable version
+
+A stable version API e.g. `v2` is considered feature complete.
+
+Any changes to the object schema do not require editing or re-creating of Custom Resources.
+Schema fields can't be removed, only new fields can be added with a default value that
+doesn't affect the controller's current behaviour.
+
+A stable version API becomes deprecated once a subsequent stable version is released.
+Stable API versions are not subject to removal in any future release of a controller major version.
```

Comment:
```suggestion
Stable API versions are not subject to removal in any future release within a controller major version.
```

Makes it clearer that we're talking about a single major version and not about the possibility that a stable API version may be removed in a subsequent major controller version.